src/python/pants/engine/target.py

```diff
@@ -868,6 +848,27 @@ def __init__(
         self.expect_single_field_set = expect_single_field_set
 
 
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class FieldSetsPerTarget(Generic[_AFS]):
```

Comment:
Why keep `ValidTargetsToFieldSets`? This seems to supersede it. This is already one of our most confusing APIs, and now we have two of them that look very similar.

Reply:
Just because removing it is a bigger change, but I'm happy to do it.

Reply:
Well, the difference is that this is dirt simple, operates on arbitrary targets that you give it, and does no validation. TargetsToValidFieldSets is semi-widely used, operates specifically on target roots, with origins, and does validation, generating error messages. That is a pretty different and more complex/specific piece of functionality. So it does make sense to retain it. My thing doesn't really supersede, it just factors out the pre-validation part of the logic.

Reply:
That said, the naming is confusing. I might rename `ValidTargetsToFieldSets` to something like `TargetRootsToFieldSets`, since that's what it really is (or even `OwnedTargetRootsToFieldSets` to be really pedantic)

Reply:
I like TargetRootsToFieldSets!

Reply:
Yeah, the "valid" isn't really carrying its weight...